internal/server/server.go

```diff
@@ -0,0 +1,126 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+	"log"
+	"os"
+
+	"github.com/Pratham700/github-search-service/internal/github"
+	"github.com/Pratham700/github-search-service/internal/util"
+	pb "github.com/Pratham700/github-search-service/proto/proto"
+)
+
+type GithubSearchServer struct {
+	pb.UnimplementedGithubSearchServiceServer
+	GitHubClient *github.GitHubClient
```

Comment:
This does not need to be public.

```suggestion
	gitHubClient *github.GitHubClient
```